designer/client/src/actions/nk/process.ts

```diff
@@ -27,12 +29,13 @@ export function fetchProcessToDisplay(processName: ProcessName, versionId?: Proc
 
 export function loadProcessState(processName: ProcessName): ThunkAction {
     return (dispatch) =>
-        HttpService.fetchProcessState(processName).then(({ data }) =>
+        HttpService.fetchProcessState(processName).then(({ data }) => {
             dispatch({
                 type: "PROCESS_STATE_LOADED",
                 processState: data,
-            }),
-        );
+            });
+            dispatch(updateEnabledCustomActions(processName));
```

Comment:
I don't understand what we want to achieve. IIUC Previously we determined the custom action enable/disable state, based on the scenario state. Could you describe in few words, we do we want to change this behavior?

Reply:
I've added some description above.  In this particular case we want to enable buttons only in case when deployed version of the scenario is visible. So after switching to old version or after creation of the new one we have to have opportunity to update state.

Reply:
Thanks for the explanation! 

Ok, so in this case, we will make two API requests every 10 seconds

- `api/processes/<scenario_name>/status`
- `api/processManagement/customAction/<scenario>}/<scenario_version>`

and `/status` send in the response

```
{
    ...
    "allowedActions": [
        "DEPLOY",
        "CANCEL"
    ],
    ...
}
```
We duplicate a logic here. Did you check the possibility of extending the existing endpoint `allowedActions` property, and for example pass the status of specific scenarios version in the `/status` endpoint? Do you know about this approach?

Also, I can see a problem with adding it to the `loadProcessState`, because, it can be an option, that we will ask for the status, only in the case of the current version (of course if we extend `/status` endpoint, it will have the sense  to ask for a specific version status)

Reply:
I've got mixed felling about that :).  From one side I was told that we wanna keep custom action as a separate logic  and separate endpoint, from the other one I have another task to make custom action configurable from the files and make it a little closer to the build-in actions (opposed to the current solution where custom action is part of the deployment manager). So from that perspective I guess I could use (or at least try to use) `api/processes/<scenario_name>/status` endpont.